### ISSUE TITLE
setTimeout can be broken by changing global.Date

### ISSUE THREAD

**Issue by glenjamin** [joyent/node#7195]
As discussed in cjohansen/Sinon.JS#419

This doesn't work in 0.10 and below
```javascript
var realSetTimeout = global.setTimeout;
var realDate = global.Date;
global.setTimeout = function fakeSTO() {}
global.Date = function fakeDate() {}

realSetTimeout(console.log, 10, "hello")

// After you run this line, the timer will fire
global.Date = realDate
```

Obviously this isn't a common use-case, but it was rather confusing until I found https://github.com/joyent/node/blob/v0.10.26/lib/timers.js#L53

0.11 is not affected by this, as it uses a different internal timing mechanism.

A relatively simply fix would be to do `var RealDate = Date;` at the top of timers.js

**Comment by mk-pmb** [comment]
This issue [should be solved](https://github.com/joyent/node/pull/3710#issuecomment-35404419) now; is it?

**Comment by cjihrig** [comment]
This is solved, post 0.10. @tjfontaine mentioned that it would be possible to solve another way in 0.10, so I think we should leave this open in case someone wants to give it a shot.

**Comment by glenjamin** [comment]
As mentioned in https://github.com/cjohansen/Sinon.JS/issues/419#issuecomment-50875873 - the timer changes were backported to the 0.10 branch, which had the nice benefit of fixing this issue.
